### PR TITLE
[Feat] dict column name

### PR DESCRIPTION
**Pull request by IHEII**
## Summary

- Support dict column name

## Solution Description

- Support dict column name, use bitmap instead of string in LS operation

### CODE REVIEW

File: src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java [hunk 1]
```diff
@@ -47,15 +51,28 @@ public ObTableOperationType getOperationType() {
...
-        switch (result.getPcode()) {
-            case Pcodes.OB_TABLE_API_EXECUTE:
-                affectedRows = ((ObTableOperationResult) result).getAffectedRows();
-                break;
-            case Pcodes.OB_TABLE_API_QUERY_AND_MUTATE:
-                affectedRows = ((ObTableQueryAndMutateResult) result).getAffectedRows();
-                break;
-            default:
-                throw new ObTableException("unknown result type: " + result.getPcode());
+        if (result instanceof ObTableSingleOpResult) {
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java:54]: could you based on pcode to return affected rows directly

File: src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java [hunk 2]
```diff
@@ -65,15 +82,28 @@ public long getAffectedRows() {
      */
     public Row getOperationRow() {
         Map<String, Object> rowsMap;
-        switch (result.getPcode()) {
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java:68]: could you based on pcode to return affected rows directly?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableBatchOperationResult.java [hunk 3]
```diff
@@ -130,5 +130,4 @@ public void addResult(ObTableOperationResult result) {
     public void addAllResults(List<ObTableOperationResult> results) {
         this.results.addAll(results);
     }
-
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableBatchOperationResult.java:133]: need remove this line?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java [hunk 4]
```diff
@@ -81,23 +107,37 @@ public Object decode(ByteBuf buf) {
         // 0. decode header
         super.decode(buf);
 
-        int len = (int) Serialization.decodeVi64(buf);
-        for (int i = 0; i < len; i++) {
-            String column = Serialization.decodeVString(buf);
-            rowKeyNames.add(column);
+        rowKeyBitLen = Serialization.decodeVi64(buf);
+        rowKeyBitMap = new byte[(int) Math.ceil(rowKeyBitLen / 8.0)];
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java:114]: could you abstract it into a function and add comment?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java [hunk 5]
```diff
@@ -176,6 +226,91 @@ public static ObTableSingleOpEntity getInstance(String[] rowKeyNames, Object[] r
...
+    /*
+     * adjustRowkeyColumnName should be execute in the last
+     */
+    public void adjustRowkeyColumnName(Map<String, Long> columnNameIdxMap) {
+        this.rowKeyBitLen = columnNameIdxMap.size();
+        int size = (int) Math.ceil(columnNameIdxMap.size() / 8.0);
+        byte[] byteArray = new byte[size];
+        List<Long> columnNameIdx = new LinkedList<>();
+
+
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java:267]: reduce a line?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java [hunk 6]
```diff
@@ -176,6 +226,91 @@ public static ObTableSingleOpEntity getInstance(String[] rowKeyNames, Object[] r
...
+        this.rowKeyBitMap = byteArray;
+    }
+
+    public void adjustPropertiesColumnName(Map<String, Long> columnNameIdxMap) {
+        if (!ignoreEncodePropertiesColumnNames) {
+            this.propertiesBitLen = columnNameIdxMap.size();
+        }
+        int size = (int) Math.ceil(columnNameIdxMap.size() / 8.0);
+        byte[] byteArray = new byte[size];
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java:298]: this two functions can abstract into one function

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    private long                  affectedRows  = 0;
+    private String                executeHost;
+    private int                   executePort;
+    private List<String>          propertiesColumnNames = new ArrayList<>();
+    /*
+     * Get pcode.
+     */
+    @Override
+    public int getPcode() {
+        return Pcodes.OB_TABLE_API_EXECUTE;
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java:41]: API_EXECUTE -> LS_EXECUTE

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+        // 0. decode version
+        super.decode(buf);
+
+        // 1. decode ObTableResult
+        this.header.decode(buf);
+
+        // 2. decode types
+        this.operationType = ObTableOperationType.valueOf(Serialization.decodeI8(buf.readByte()));
+
+        // 3. decode Entity TODO 需要check一下errno
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java]: check TODO?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+        return this;
+    }
+
+    /*
+     * Get payload content size.
+     */
+    @Override
+    public long getPayloadContentSize() {
+        return 1 + header.getPayloadSize() + entity.getPayloadSize()
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java:107]: use constant ?

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableTabletOpResult.java [hunk 10]
```diff
@@ -17,5 +17,113 @@
...
+import static com.alipay.oceanbase.rpc.util.Serialization.getObUniVersionHeaderLength;
+
+public class ObTableTabletOpResult extends AbstractPayload {
+    private List<ObTableSingleOpResult> results = new ArrayList<ObTableSingleOpResult>();
+    private List<String> propertiesColumnNames = new ArrayList<>();
+
+
+    @Override
+    public int getPcode() {
+        return Pcodes.OB_TABLE_API_BATCH_EXECUTE;
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableTabletOpResult.java:38]: BATCH_EXECUTE -> LS_EXECUTE

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableBatchOperationResult.java (hunk 3 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableBatchOperationResult.java:133]: For the format

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java (hunk 6 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java:289]: Fixed, set into `isSamePropertiesColumnNamesLen()`

File: src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java (hunk 1 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java]: Fixed

File: src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java (hunk 2 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/mutation/result/MutationResult.java]: Fixed

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java (hunk 4 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java]: Fixed, has set into `isSamePropertiesColumnNamesLen()`

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java (hunk 5 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpEntity.java]: Fixed

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java (hunk 7 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java]: Fixed

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java (hunk 8 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java]: Fixed

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java (hunk 9 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableSingleOpResult.java]: Fixed, have commented

File: src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableTabletOpResult.java (hunk 10 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/protocol/payload/impl/execute/ObTableTabletOpResult.java]: Fixed

**Comment by WeiXinChan** [comment]
LGTM

**Pull request merged**